### PR TITLE
[Documentation] Add README to R-C-V

### PR DESCRIPTION
**Pull request by philipfweiss**
Adds basic documentation of R-C-V Api. Very rough - please critique!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,67 @@
-# react-component-variations
+# React-Component-Variations
+**React-Component-Variations** is a library for traversing data from a project's component variations.   
 
-Warning: this project is still very alpha.
\ No newline at end of file
+**What is a Component Variation?**  
+Airbnb uses Storybook (https://github.com/storybooks/storybook) to iteratively test and develop react components, as well as for documentation. We use an abstraction called a *variation* on top of Storybook, which logically represents the specific usages of a more general component inside of our codebase. For example, in [React-Dates](http://airbnb.io/react-dates/?selectedKind=DayPicker&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel), the DayPicker component has variations for when the component has no border, when it has an info panel, when it is vertical, etc. Variations help engineers correctly use the abstractions built into your components by providing examples of concrete usage.
```
Comment (majapw) [inline README.md]: I'm not sure that this aligns with @ljharb's vision for this library so I'll let him comment. :)

**Comment by ahuth** [comment]
Closing. I believe we don't need this anymore as of #19 .

**Comment by ljharb** [comment]
We do; this documentation is more about writing consumers.

**Comment by ahuth** [comment]
I'm writing documentation on that. After almost a year of this PR being opened, it seems perhaps unlikely that it's going in.

But if you want to keep it around that's okay

**Comment by ljharb** [comment]
That’s because i was intentionally not adding public documentation yet.

Perhaps we should collaborate on this directly before making further PRs?

**Comment by madicap** [comment]
@ljharb If you'd be willing to tell me (through email or whatever other channel), I'd be interested in hearing more about why you're strategically avoiding adding public documentation

**Comment by ljharb** [comment]
It’s not publicly usable yet - when all the consumers are available, and it can be introduced as a complete system, it will be ready for public documentation. In the meantime, external users might be falsely encouraged to adopt variations before they’re ready, and when there still might be major changes (like the import codemod we had to do, for example).

**Comment by ahuth** [comment]
I gotcha 👍

**Comment by ljharb** [comment]
It’d probably be helpful to set up a call week after next so we can all get on the same page about the overall strategy and plan for r-c-v :-)

**Comment by ahuth** [comment]
Sounds good. I'll send you an email

**Pull request closed without merging**